tools/cli/adminClusterCommands.go

```diff
@@ -36,10 +37,15 @@ import (
 
 // An indirection for the prompt function so that it can be mocked in the unit tests
 var promptFn = prompt
+var validSearchAttributeKey = regexp.MustCompile(`^[a-zA-Z][a-zA-Z_0-9]*$`)
 
 // AdminAddSearchAttribute to whitelist search attribute
 func AdminAddSearchAttribute(c *cli.Context) {
 	key := getRequiredOption(c, FlagSearchAttributesKey)
+	if !isValidSearchAttributeKey(key) {
+		ErrorAndExit("Invalid search-attribute key.", nil)
```

Comment:
Could you provide more context here with what is wrong with the key? It is not obvious what it allowed and what failed the validation.